main/src/test/java/org/mobilitydata/gtfsvalidator/validator/TripTravelSpeedValidatorTest.java

```diff
@@ -0,0 +1,443 @@
+    TripTravelSpeedValidator underTest = new TripTravelSpeedValidator();
+    underTest.stopTable =
+        createStopTable(
+            noticeContainer,
+            ImmutableList.of(
+                createStop(2, "s0", 16.251233d, -61.591997d),
+                createStop(3, "s1", 16.251053d, -61.591241d),
+                createStop(4, "s2", 16.242191d, -61.588554d)));
+
+    underTest.tripTable = createTripTable(noticeContainer, ImmutableList.of(createTrip()));
+
+    underTest.stopTimeTable =
+        createStopTimeTable(
+            noticeContainer,
+            ImmutableList.of(
+                createStopTime(
+                    4,
+                    "t1",
+                    "s0",
+                    0,
+                    GtfsTime.fromSecondsSinceMidnight(130),
+                    GtfsTime.fromSecondsSinceMidnight(100)),
+                createStopTime(5, "t1", "s1", 1, GtfsTime.fromSecondsSinceMidnight(131), null),
+                createStopTime(
+                    6,
+                    "t1",
+                    "s2",
+                    4,
+                    GtfsTime.fromSecondsSinceMidnight(360),
+                    GtfsTime.fromSecondsSinceMidnight(140))));
+
+    underTest.validate(noticeContainer);
+
+    assertThat(noticeContainer.getValidationNotices())
+        .containsExactly(new FastTravelBetweenStopsNotice("t1", 443.7847945712592d, 0, 4));
+  }
+
+  @Test
+  public void noFastTravelMissingArrivalTimeShouldNotGenerateNotice() {
+    NoticeContainer noticeContainer = new NoticeContainer();
+    TripTravelSpeedValidator underTest = new TripTravelSpeedValidator();
+    underTest.stopTable =
+        createStopTable(
+            noticeContainer,
+            ImmutableList.of(
+                createStop(2, "s0", 16.251233d, -61.591997d),
+                createStop(3, "s1", 16.251053d, -61.591241d),
+                createStop(4, "s2", 16.242191d, -61.588554d)));
+
+    underTest.tripTable = createTripTable(noticeContainer, ImmutableList.of(createTrip()));
+
+    underTest.stopTimeTable =
+        createStopTimeTable(
+            noticeContainer,
+            ImmutableList.of(
+                createStopTime(
+                    4,
+                    "t1",
+                    "s0",
+                    0,
+                    GtfsTime.fromSecondsSinceMidnight(130),
+                    GtfsTime.fromSecondsSinceMidnight(100)),
+                createStopTime(5, "t1", "s1", 1, GtfsTime.fromSecondsSinceMidnight(131), null),
+                createStopTime(
+                    6,
+                    "t1",
+                    "s2",
+                    4,
+                    GtfsTime.fromSecondsSinceMidnight(770000),
+                    GtfsTime.fromSecondsSinceMidnight(390000))));
+
+    underTest.validate(noticeContainer);
+
+    assertThat(noticeContainer.getValidationNotices()).isEmpty();
+  }
+
+  @Test
+  public void fastTravelMissingDepartureTimeShouldGenerateNotice() {
```

Comment:
We need another two tests where both arrival and departure time aren't provided in between stop times that have arrival and departure times, both for the case where there is and is not too fast travel. The below is valid legacy GTFS if the "timepoint" field isn't provided.

|  stop_sequence | arrival_time | departure_time |
|  ------ | ------ | ------ | 
1 | 00:00:00 | 00:00:00  |  
2 |  |  |  
3 |  |  |  
4 | 00:10:00 | 00:10:00  |  
5 |  |  |  
6 | 00:20:00 | 00:20:00  |